src/metax_api/tests/api/rest/base/views/directories/read.py

```diff
@@ -612,6 +675,26 @@ def test_browsing_in_cr_context(self):
         response = self.client.get('/rest/directories/3/files?cr_identifier=%s' % cr_pk)
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
+    @responses.activate
+    def test_browsing_in_not_cr_context(self):
+        '''
+        Cr with open access type should be available for any end-user api user. Browsing files for a cr with restricted
+        access type should be forbidden for non-owner (or service) user.
+        '''
+        self._use_http_authorization(method='bearer', token=self.token)
+        response = self.client.get('/rest/directories/3/files?not_cr_identifier=1')
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        self._set_http_authorization('service')
+        response = self.client.get('/rest/datasets/1')
+        json = response.data
+        json['research_dataset']['access_rights']['access_type']['identifier'] = ACCESS_TYPES['restricted']
+        response = self.client.put('/rest/datasets/1', response.data, format='json')
+
+        self._use_http_authorization(method='bearer', token=self.token)
+        response = self.client.get('/rest/directories/3/files?not_cr_identifier=1')
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
```

Comment:
The use case for ``not_cr_identifier`` is specifically to help pick files when editing a dataset, so the last block about owner not being able to browse files with ``not_cr_identifier`` if access type is restricted is problematic.

Not directly related to the test, but a general tip: In the middle block, instead of using the API to prepare the data being tested, you can also directly use ``cr = CR.objects.get(pk=1)`` and change its fields and save using ``cr.force_save()``. Might save some typing.